tests/test_nml_utils.py

```diff
@@ -9,31 +12,59 @@ def test_ensure_max_edge_length():
     # test the graph version
     test_nml_graph = generate_graph(test_nml)
     test_result_nml_graph = ensure_max_edge_length(test_nml_graph, max_length)
+    assert_max_length(test_result_nml_graph, max_length)
```

Comment:
I think it makes sense to add another check which makes sure that the assertion is violated before running the `ensure_max_edge_length` code. Otherwise, the code would pass if the original NML already passes the assertion. However, we explicitly want to test that an NML which doesn't passes the assertion is successfully processed.